### PR TITLE
Allow slices and integers as a value for arguments in the DefinitionImage struct

### PR DESCRIPTION
**Pull request by pbusko**
## Summary

This PR introduces backward compatible support for lists of strings/integers as a value type for the docker build arguments.

## Use Cases

This opens a possibility to integrate stack's image dependencies to be managed by tools like [Renovate](https://docs.renovatebot.com/) and it's `repology` datasource. For example, the `stack.toml` can be changed in the following way:

```toml
  [run.args]
    packages = [
      # renovate: depName=bash
      "bash=5.0-6ubuntu1",
      # renovate: depName=dash
      "dash=0.5.11+git20210903+057cd650a4ed-3build1",
      # renovate: depName=ca-certificates
      "ca-certificates=20211016",
      # renovate: depName=glibc
      "libc6=2.35-0ubuntu3",
      # renovate: depName=gcc-12
      "libgcc-s1=12.0.0-2ubuntu1~22.04",
      # renovate: depName=openssl
      "libssl3=3.0.2-0ubuntu1.6",
      # renovate: depName=gcc-12
      "libstdc++6=12.0.0-2ubuntu1~22.04",
      # renovate: depName=zlib
      "zlib1g=1:1.2.11.dfsg-2ubuntu9"
    ]
```

And configure `regexManager` in the renovate config:

```json
  "regexManagers": [
    {
      "fileMatch": [
        "^stack\\/stack\\.toml$"
      ],
      "matchStrings": [
        "# renovate: depName=(?<depName>.*?)\\s+.*?\\=(?<currentValue>.*?)\""
      ],
      "datasourceTemplate": "repology",
      "versioningTemplate": "loose",
      "depNameTemplate": "ubuntu_22_04/{{{depName}}}"
    }
  ],
```

With the changes above, it's no longer necessary to perform scheduled builds, since all the dependencies are of a specific version with bumps coming through PRs from renovate.

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: internal/ihop/definition.go
Comment (robdimsdale) [inline internal/ihop/definition.go:108]: I assume that this is a typo and should read: `typedSlice`?

**Comment by robdimsdale** [comment]
I'm happy with this, but I'd like other stacks maintainers (and @ryanmoran ) to take a look first. I have a couple of inline comments but no major issues.

I don't see any performance concerns with type introspection, and I don't see any other issues with this approach.

As an aside, I'd love to learn more about you're using Renovate + repology to see if we can leverage that in our stacks. It seems like this approach might enable us to pin packages to specific versions in the `stack.toml` and yet still leverage automatic updates to these versions, which would be a huge improvement over what we have today. Is there an example repo you can point us to?

File: internal/ihop/definition.go
Comment (pbusko) [reply internal/ihop/definition.go]: yup, typo has been fixed :)

File: internal/ihop/definition.go
Comment (pbusko) [reply internal/ihop/definition.go:116]: good catch, I've added missing test case

**Comment by sophiewigmore** [comment]
I'm in agreement with @robdimsdale that the changes seem reasonable, are well tested, and I've confirmed everything definitely works in a backwards-compatible manner when I manually tested everything out

**Review by sophiewigmore — DISMISSED** [review]

**Comment by pbusko** [comment]
I've forked the jammy tiny stack repository to demonstrate, how Renovate can handle dependency updates of the ubuntu packages (and also github workflows/go dependencies):
https://github.com/pbusko/jammy-tiny-stack

The most important thing is [renovate.json](https://github.com/pbusko/jammy-tiny-stack/blob/main/renovate.json) (more information on the [official documentation website](https://docs.renovatebot.com/))

As you can see, renovate is able to create PRs with updated to the base docker images (`FROM` statements), and the ubuntu packages (e.g. https://github.com/pbusko/jammy-tiny-stack/pull/24 and https://github.com/pbusko/jammy-tiny-stack/pull/25)
Also it provides you with a handy [dependency dashboard](https://github.com/pbusko/jammy-tiny-stack/issues/22)

**Note**: the author of the issue and the PRs is myself, due to the fact that I triggered renovate locally as a CLI with my own token

**Note**: The public GitHub App currently experiencing issues with the `repology` datasource due to the rate limits. See more info here https://github.com/renovatebot/renovate/issues/18144. Apart from using the official GitHub App, renovate offers [different ways for self-hosting](https://github.com/renovatebot/renovate/blob/main/docs/usage/getting-started/running.md) (can be easily integrated into basically any CI system)

**Comment by sophiewigmore** [comment]
@pbusko thank you for explaining that, how cool! This is super useful and could definitely be useful in our project too

**Comment by pbusko** [comment]
> @pbusko thank you for explaining that, how cool! This is super useful and could possibly be used in our project too

Glad to hear! As a side note, we're also using renovate for managing entire meta buildpack and builder dependencies (can be achieved by using regex manager + github releases datasource)

**Review by sophiewigmore — APPROVED** [review]

**Comment by loewenstein** [comment]
@ryanmoran could you please have a look. I think we already lost @sophiewigmore's approval due to required rebases.

Also, could you imagine a timely release, has been nearly a month since the last one.

**Review by ryanmoran — APPROVED** [review]

**Comment by robdimsdale** [comment]
Ok we have all the approvals so once this is rebased (I can't do this for you) we can merge.

I know the rebasing will invalidate the reviews, but I'll add my review and merge first thing tomorrow.

**Pull request merged**
